### ISSUE TITLE
Better comments and docstrings

### ISSUE THREAD

**Issue by divad-nhok** [cgre-aachen/gempy#189]
(1) Problem summary: documentation, comments, docstrings and examples are not clear in describing what is going on in various classes, functions, method. See "(4) Full problem description" for more info.

(2) Solution: clearer documentation and comments; at the very least, informative docstrings to make clear what parameters and arguments expected in methods and what the methods are doing.

(3) Alternatives: the alternative is fully reading and understanding the code, which is doable, but a significant time investment. I feel it would be easier for someone who wrote the code to explain what's going on rather than me following every line of code logic.

(4) Full problem description:

I've been trying to understand the notebook gempy/notebooks/tutorials/ch2-1-Gravity.ipynb.

(a) I was trying to understand the following line in the notebook: geo_model.set_gravity_interpolator(pos_density=1)
This refers to the specific method gempy/core/model.py >> class Model >> method set_gravity_interpolator
I feel set_gravity_interpolator could use a docstring to help users understand what the density_block and pos_density parameters are doing.
In trying to understand where the density values are coming from, I have been trying to parse the comment (model.py line 917), "If you do not pass the density block you need to pass the position of surface values where density is assigned" and am unclear on a number of things. What is the "density block", is it the measured density at each sensor as defined for the gravity block or something else? What is "the position of surface values where density is assigned"? What are the surface values and what does position refer to for the surface values? I'm having seeing where the density values are even read into the example notebook.

(b) I’d like to know (in summary/broadly) what specific steps/method or function calls are taking place in geo_model.set_gravity_interpolator and gp.set_interpolation_data. I’m finding the specific steps of the code a bit hard to follow at the moment (without spending a chunk of time reading through all the code logic and function calls.

**Comment by divad-nhok** [comment]
Adding on to the pos_density component of this.
I can see that if pos_density equals 1, then density is eventually set after a call to InterpolatorGravity and then the assignment, density = self.theano_graph.compute_series()[0][pos_density, :- 2 * self.theano_graph.len_points], which itself calls the compute_series property of that theano_graph object. Like I can sort of see what compute_series is doing now that I have a look at it, but it's been too much work, for someone who just wants to use the set_gravity_interpolator function, to get to this point.

**Comment by Leguark** [comment]
yes, nobody should be forced to go all the way to the theano hell to understand what is going on. Not many are able to come back :) Sorry for the inconvenience.  

(1-3) Yes, we fully agree we are working on that and hopefully soon the community will start to add several examples they are working on. In special gravity is lacking on documentation and api. My plans is to finish the docstrings in the next 2-3 weeks

(a) #190 As I explained in the other issue we can add values to each layer to populate them. Since the forward gravity I also compute it in theano (what is only useful for automatic differentiation and probably not worth doing it) I need to pass to theano which column of the "interpolated values" is the density to perform gravity. The idea is to be able to pass a string with the proper name of the property (i.e. 'density' in this case) but it is not implemented yet.

As I mentioned in another issue, in my opinion once you assign a physical property to each voxel and you get the correspondent numpy array in the Out of gempy.compute_model. I will first prototype the mag forgetting about theano.

Remember that the XYZ values for each lith_block are stored on geo_model.grid.gravity_grid (this name we totally need to refactor it to something more general). The different grids and what mean to activate and deactivate them also needs a lot of explanation.
